Core/StitchCoreSDK/Sources/StitchCoreSDK/StitchError.swift

```diff
@@ -110,6 +110,7 @@ public enum StitchClientErrorCode {
     case couldNotLoadPersistedAuthInfo
     case couldNotPersistAuthInfo
     case couldNotLoadSyncInfo
+    case couldNotCommunicateWithDriver(withError: Error)
```

Comment:
[naming minor] this is not necessarily a communication error. it could be an error with the query (like a duplicate key error). This could probably be more accurately named `mongoDriverError`